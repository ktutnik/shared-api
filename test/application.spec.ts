import supertest from "supertest"
import createApp from "../src/app"
import { closeConnection, createUser, ignoreProps } from "./helper"

afterEach(async () => {
    await closeConnection()
})

describe("Application", () => {
    it("Should create application properly, should not active by default", async () => {
        const app = await createApp({ mode: "production" })
        const user = await createUser(app)
        const { body } = await supertest(app.callback())
            .post("/api/backend/applications")
            .set("Authorization", `Bearer ${user.token}`)
            .send({ name: "Lorem Ipsum" })
            .expect(200)
        const { body: result } = await supertest(app.callback())
            .get(`/api/backend/applications/${body.id}`)
            .set("Authorization", `Bearer ${user.token}`)
            .expect(200)
        expect(result).toMatchSnapshot({ ...ignoreProps, apiKey: expect.any(String) })
    })
    it("Should able to activate basic application", async () => {
        const app = await createApp({ mode: "production" })
        const user = await createUser(app)
        const { body } = await supertest(app.callback())
            .post("/api/backend/applications")
            .set("Authorization", `Bearer ${user.token}`)
            .send({ name: "Lorem Ipsum" })
            .expect(200)
        await supertest(app.callback())
            .post(`/api/backend/applications/${body.id}/activate`)
            .set("Authorization", `Bearer ${user.token}`)
            .send({ payment: 5 })
            .expect(200)
        const { body: result } = await supertest(app.callback())
            .get(`/api/backend/applications/${body.id}`)
            .set("Authorization", `Bearer ${user.token}`)
            .expect(200)
        expect(result).toMatchSnapshot({ ...ignoreProps, apiKey: expect.any(String) })
    })
    it("Should able to activate premium application", async () => {
        const app = await createApp({ mode: "production" })
        const user = await createUser(app)
        const { body } = await supertest(app.callback())
            .post("/api/backend/applications")
            .set("Authorization", `Bearer ${user.token}`)
            .send({ name: "Lorem Ipsum" })
            .expect(200)
        await supertest(app.callback())
            .post(`/api/backend/applications/${body.id}/activate`)
            .set("Authorization", `Bearer ${user.token}`)
            .send({ payment: 20 })
            .expect(200)
        const { body: result } = await supertest(app.callback())
            .get(`/api/backend/applications/${body.id}`)
            .set("Authorization", `Bearer ${user.token}`)
            .expect(200)
        expect(result).toMatchSnapshot({ ...ignoreProps, apiKey: expect.any(String) })
    })
})