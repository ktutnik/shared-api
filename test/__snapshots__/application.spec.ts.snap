// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application Should able to activate basic application 1`] = `
Object {
  "active": true,
  "apiKey": Any<String>,
  "createdAt": Any<String>,
  "deleted": false,
  "id": Any<Number>,
  "name": "Lorem Ipsum",
  "type": "Basic",
  "updatedAt": Any<String>,
}
`;

exports[`Application Should able to activate premium application 1`] = `
Object {
  "active": true,
  "apiKey": Any<String>,
  "createdAt": Any<String>,
  "deleted": false,
  "id": Any<Number>,
  "name": "Lorem Ipsum",
  "type": "Premium",
  "updatedAt": Any<String>,
}
`;

exports[`Application Should create application properly, should not active by default 1`] = `
Object {
  "active": false,
  "apiKey": Any<String>,
  "createdAt": Any<String>,
  "deleted": false,
  "id": Any<Number>,
  "name": "Lorem Ipsum",
  "type": "Basic",
  "updatedAt": Any<String>,
}
`;
